lib/ecto/query/planner.ex

```diff
@@ -676,6 +680,62 @@ defmodule Ecto.Query.Planner do
     %{query | wheres: wheres, havings: havings}
   end
 
+  @spec plan_order_bys(Ecto.Query.t(), module, map()) :: Ecto.Query.t
+  defp plan_order_bys(query, adapter, cte_names) do
+    order_bys =
+      Enum.map(query.order_bys, fn
+        %{subqueries: []} = order_by ->
+          order_by
+
+          %{subqueries: subqueries} = order_by ->
+          %{order_by | subqueries: Enum.map(subqueries, &plan_subquery(&1, query, nil, adapter, false, cte_names))}
+      end)
```

Comment:
We can probably extract this into a shared function. Perhaps we can extract the whole `plan_order_bys` function into a `plan_by(:order_by, query, adapter, cte_names)`.

Reply:
The only "problem" with that is the subtly different state for `order_by`, `distinct`, `group_by` and `window`.

Actually I can do it for `order_by` and `group_by` as those actually are the same, but windows is a keyword list and `distinct` is a single `ByExpr`.

Reply:
I've done it for the two that seemed reasonable.